### PR TITLE
Module 6: Data persistence

### PR DESCRIPTION
**Pull request by AlejoArea**
Added floor database, added provider, added splashscreen, added favicon and changed the name of the app natively.

https://github.com/AlejoArea/TheMovieDB/assets/114313724/0b4d80f9-8099-46ff-aaa8-82accc56863f

### CODE REVIEW

File: lib/src/domain/usecase/implementation/usecase_genres.dart [hunk 1]
```diff
@@ -1,18 +1,45 @@
...
-    return allGenres;
+  Future<DataState<List<Genre>>> repositoryCall(
+      [CategoryEnum? category]) async {
+    final ConnectivityResult connectivityResult =
+        await (connectivity.checkConnectivity());
+    late DataState<List<Genre>> allGenres;
+    late List<Genre>? genres;
+    if (connectivityResult == ConnectivityResult.none) {
+      genres = await genreDatabaseRepository.getAllGenres();
+      allGenres = DataState(data: genres, state: DataEvents.success);
```
Comment (FranciscoGaraventa) [inline lib/src/domain/usecase/implementation/usecase_genres.dart]: trailing comma

File: lib/src/data/repository/movie_database_repository.dart [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import 'package:themoviedb/src/data/datasource/local/movie_database.dart';
+import 'package:themoviedb/src/domain/entity/movie.dart';
+import 'package:themoviedb/src/domain/repository/i_database_repository.dart';
```
Comment (FranciscoGaraventa) [inline lib/src/data/repository/movie_database_repository.dart]: Relative imports

File: lib/src/data/repository/genre_database_repository.dart [hunk 3]
```diff
@@ -0,0 +1,23 @@
+
+import 'package:themoviedb/src/domain/repository/i_database_repository.dart';
```
Comment (FranciscoGaraventa) [inline lib/src/data/repository/genre_database_repository.dart]: relative import

File: lib/src/domain/usecase/implementation/usecase_movie.dart [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        return DataState(
+          data: [],
+          state: DataEvents.empty,
+        );
+      } else {
+        return DataState(
+          data: movies,
+          state: DataEvents.success,
+        );
+      }
```
Comment (FranciscoGaraventa) [inline lib/src/domain/usecase/implementation/usecase_movie.dart]: Here you can made only on DataState combining both results like
```
return DataState(
          data: movies!.isEmpty ? [] : movies,
          state: movies!.isEmpty ? DataEvents.empty : DataEvents.empty,
);
```

File: lib/src/presentation/bloc/bloc_popular.dart [hunk 5]
```diff
@@ -1,5 +1,7 @@
 import 'dart:async';
 
+import 'package:themoviedb/src/core/util/enum_category.dart';
```
Comment (FranciscoGaraventa) [inline lib/src/presentation/bloc/bloc_popular.dart]: relative import

File: lib/src/presentation/bloc/bloc_top_rated.dart [hunk 6]
```diff
@@ -1,5 +1,7 @@
 import 'dart:async';
 
+import 'package:themoviedb/src/core/util/enum_category.dart';
```
Comment (FranciscoGaraventa) [inline lib/src/presentation/bloc/bloc_top_rated.dart]: relative import

File: lib/src/presentation/bloc/bloc_upcoming.dart [hunk 7]
```diff
@@ -1,5 +1,7 @@
 import 'dart:async';
 
+import 'package:themoviedb/src/core/util/enum_category.dart';
```
Comment (FranciscoGaraventa) [inline lib/src/presentation/bloc/bloc_upcoming.dart]: relative import

Comment (AlejoArea) [reply lib/src/presentation/bloc/bloc_upcoming.dart]: Done

File: lib/src/presentation/bloc/bloc_top_rated.dart (hunk 6 above)
Comment (AlejoArea) [reply lib/src/presentation/bloc/bloc_top_rated.dart]: done

File: lib/src/presentation/bloc/bloc_popular.dart (hunk 5 above)
Comment (AlejoArea) [reply lib/src/presentation/bloc/bloc_popular.dart]: done

File: lib/src/data/repository/genre_database_repository.dart (hunk 3 above)
Comment (AlejoArea) [reply lib/src/data/repository/genre_database_repository.dart]: done

File: lib/src/data/repository/movie_database_repository.dart (hunk 2 above)
Comment (AlejoArea) [reply lib/src/data/repository/movie_database_repository.dart]: done

File: lib/src/domain/usecase/implementation/usecase_genres.dart (hunk 1 above)
Comment (AlejoArea) [reply lib/src/domain/usecase/implementation/usecase_genres.dart]: done

File: lib/src/domain/usecase/implementation/usecase_movie.dart (hunk 4 above)
Comment (AlejoArea) [reply lib/src/domain/usecase/implementation/usecase_movie.dart]: Done!

File: lib/src/data/model/movies_model.dart [hunk 8]
```diff
@@ -13,23 +13,24 @@ class MovieModel {
...
+      required this.releaseDate,
+      required this.overview,
+      required this.voteAverage,
+      required this.genres,
+      required this.posterPath,
+      required this.backdropPath,
+      required this.popularity,
+      required this.video,
+      required this.voteCount,
+      required this.categories});
```
Comment (Kevin-Guillen) [inline lib/src/data/model/movies_model.dart]: trailing comma

File: lib/src/domain/usecase/implementation/usecase_genres.dart [hunk 9]
```diff
@@ -1,18 +1,45 @@
+import 'package:connectivity_plus/connectivity_plus.dart';
+import 'package:themoviedb/src/core/util/enum_category.dart';
```
Comment (Kevin-Guillen) [inline lib/src/domain/usecase/implementation/usecase_genres.dart]: relative imports

**Review by Kevin-Guillen — APPROVED** [review]

**Review by CIFUM007 — APPROVED** [review]

**Review by MarcosCGlobant — APPROVED** [review]

**Pull request merged**